File: SymmetricTree/first.py
class Solution:
    # @param {TreeNode} root
    # @return {boolean}
    def helper(self, a, b):
        if a is None and b is None:
            return True
        if a is None and b is not None:
            return False
        if a is not None and b is None:
            return False
        if a.val != b.val: 
            return False
        return self.helper(a.left, b.right) and self.helper(a.right,b.left)
    def isSymmetric(self, root):
        if root is None:
            return True
        return self.helper(root.left, root.right)

File: BinaryTreeLevelOrderTraversalII/first.cpp
public class Solution {

public int level=0;
public List<List<Integer>> ans = new ArrayList<>();

 public void Build(TreeNode root) {
   if(root == null) return;
    level++;
    List<Integer> temp;
   if(level>ans.size()){
        temp=new ArrayList<>();
        temp.add( root.val);
        ans.add(0,temp);
    }
    else{
        temp=ans.get(ans.size()-level);
        temp.add(root.val);
    }

    Build(root.left);
    Build(root.right);

    level--;
  }

public List<List<Integer>> levelOrderBottom(TreeNode root) {
        Build(root);
        return ans;
  }
}
